File: apms-backend/src/main/java/com/projectmanagement/service/WorkflowService.java
package com.projectmanagement.service;

import com.projectmanagement.dto.WorkflowHistoryResponse;
import com.projectmanagement.dto.WorkflowResponse;
import com.projectmanagement.dto.WorkflowStageResponse;
import com.projectmanagement.entity.Project;
import java.util.List;

public interface WorkflowService {
    WorkflowResponse createWorkflow(Project project);
    WorkflowResponse submitWorkflow(Long projectId);
    WorkflowResponse reviewWorkflow(Long projectId, String remarks);
    WorkflowResponse approveWorkflow(Long projectId, String remarks);
    WorkflowResponse rejectWorkflow(Long projectId, String remarks);
    WorkflowResponse getWorkflow(Long projectId);
    List<WorkflowHistoryResponse> getWorkflowHistory(Long projectId);
    WorkflowStageResponse getCurrentStage(Long projectId);
    void advanceStage(Long projectId);
    void activateNextStage(Long projectId);
    void completeWorkflow(Long projectId);
}

File: apms-frontend/src/services/workflowService.js
import api from './api';

export const workflowService = {
  getWorkflow: async (projectId) => {
    const response = await api.get(`/workflows/project/${projectId}`);
    return response?.data?.data ?? response.data;
  },

  getWorkflowHistory: async (projectId) => {
    const response = await api.get(`/workflows/history/${projectId}`);
    return response?.data?.data ?? response.data;
  },

  submitWorkflow: async (projectId) => {
    const response = await api.post('/workflows/submit', { projectId });
    return response?.data?.data ?? response.data;
  },

  reviewWorkflow: async (projectId, remarks = '') => {
    const response = await api.post('/workflows/review', { projectId, remarks });
    return response?.data?.data ?? response.data;
  },

  approveWorkflow: async (projectId, remarks = '') => {
    const response = await api.post('/workflows/approve', { projectId, remarks });
    return response?.data?.data ?? response.data;
  },

  rejectWorkflow: async (projectId, remarks) => {
    const response = await api.post('/workflows/reject', { projectId, remarks });
    return response?.data?.data ?? response.data;
  },
};
